src/app/core/services/repo-url-cache.service.ts

```diff
@@ -23,6 +23,12 @@ export class RepoUrlCacheService {
     }
   }
 
+  removeFromSuggestions(repo: string): void {
+    const repoIndex = this.suggestions.indexOf(repo);
+    this.suggestions.splice(repoIndex, 1);
```

Comment:
I'm not sure if we are always guaranteed to find a result (without manual modification to the cache),
but to ensure safety, just return immediately if the repo is not found in the cache.

<img width="398" alt="image" src="https://github.com/user-attachments/assets/19f95836-b3c3-44bf-9c4e-71e4f2b8ca0e" />


This is because splice can work on negative numbers as seen above.
Thus, it is good if we just exit the function straight away if the repo didnt exist in cache.

Reply:
Might be good to include this as a test case for this file.